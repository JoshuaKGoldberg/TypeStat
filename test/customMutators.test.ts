import path from "node:path";
import { describe, expect, it } from "vitest";

import { runMutationTest } from "../src/tests/testSetup.js";

describe("Custom mutators", () => {
	it("empty array", async () => {
		const caseDir = path.join(
			import.meta.dirname,
			"cases/custom mutators/empty",
		);
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("one mutator", async () => {
		const caseDir = path.join(import.meta.dirname, "cases/custom mutators/one");
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});

	it("two mutators", async () => {
		const caseDir = path.join(import.meta.dirname, "cases/custom mutators/two");
		const { actualContent, expectedFilePath, options } =
			await runMutationTest(caseDir);
		await expect(actualContent).toMatchFileSnapshot(expectedFilePath);
		expect(options).toMatchSnapshot("options");
	});
});
