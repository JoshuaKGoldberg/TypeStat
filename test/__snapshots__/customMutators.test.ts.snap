// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Custom mutators > empty array > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "files": [
      "actual.ts"
    ],
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": false
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": false,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "mutators": [
    [
      "fixImportExtensions",
      null
    ],
    [
      "fixIncompleteTypes",
      null
    ],
    [
      "fixMissingProperties",
      null
    ],
    [
      "fixNoImplicitAny",
      null
    ],
    [
      "fixNoImplicitThis",
      null
    ],
    [
      "fixNoInferableTypes",
      null
    ],
    [
      "fixStrictNonNullAssertions",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {}
}"
`;

exports[`Custom mutators > one mutator > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "include": [
      "<rootDir>/actual.ts"
    ],
    "compileOnSave": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": false
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": false,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "include": [
    "<rootDir>/actual.ts"
  ],
  "mutators": [
    [
      "./sampleMutator.cjs",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {}
}"
`;

exports[`Custom mutators > two mutators > options 1`] = `
"{
  "cleanups": {
    "suppressTypeErrors": false
  },
  "compilerOptions": {
    "files": [
      "actual.ts"
    ],
    "noImplicitAny": false,
    "noImplicitThis": false,
    "strictNullChecks": false
  },
  "files": {
    "above": "",
    "below": "",
    "renameExtensions": false
  },
  "filters": [],
  "fixes": {
    "importExtensions": false,
    "incompleteTypes": false,
    "missingProperties": false,
    "noImplicitAny": false,
    "noImplicitThis": false,
    "noInferableTypes": false,
    "strictNonNullAssertions": false
  },
  "hints": {
    "react": {
      "propTypes": "whenRequired",
      "propTypesOptionality": "asWritten"
    }
  },
  "mutators": [
    [
      "./mutator1.cjs",
      null
    ],
    [
      "./mutator2.cjs",
      null
    ]
  ],
  "output": {},
  "package": {
    "directory": "<rootDir>",
    "file": "<rootDir>/package.json"
  },
  "postProcess": {
    "shell": []
  },
  "projectPath": "<rootDir>/tsconfig.json",
  "types": {}
}"
`;
